### PR TITLE
Improve 'TODO' filter

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/16124

### CODE REVIEW

**Comment by coolo** [comment]
this sounds like a huge performance hit. I.e. you're doing a single query for every softfail - this must be around 1000 SQL queries for the dashboard. softfails are frequent

**Comment by Martchus** [comment]
Yes. Maybe just apply the additional condition to the softfails which would be actually displayed?

**Comment by Martchus** [comment]
Not sure how to implement this faster without putting too much effort in it (eg. changing database). Maybe the first and the last commit (5c06041 and 7984956) can be cherry picked.

File: templates/test/overview.html.ep [hunk 1]
```diff
@@ -52,7 +52,7 @@
     </div>
     <div class="panel panel-default" id="filter-panel">
         <div class="panel-heading"><strong>Filter</strong>
-            <%= help_popover('Help for the <i>test overview</i>' => '
+            <%= help_popover('Help for the <em>test overview</em>' => '
```
Comment (okurz) [inline templates/test/overview.html.ep]: why?

**Comment by coolo** [comment]
we're facing the fundamental problem of the intersex nature of the softfails. Because you expect this state to be treated differently depending on the context. *This* is the underlying problem.

And what we really should do is rename one of them. Suggestions?

**Comment by okurz** [comment]
IMHO the easy part is the "soft failure with no failed modules". It means "only known failures for which we know a workaround and hopefully our workflow process is good enough that these failures are tracked elsewhere and properly referenced".

The "soft fails with failed modules" mean something like "Depending on whoever decided that a module failed but we do not want to consider the whole job failure as severe as for a failed job". That is very much depending on the environment, meaning of the individual modules and how these should be handled and what their impact is. If it would be for me then this category would maybe not exist at all as it is easy to miss critical issues overlooking these softfails, e.g. bsc#990384, "btrfs filesystem turns R/O" where the symptom was just one module "yast_i" failing turning the job softfailed with "just" one failing module. I don't know how to handle these. To me they are basically to be treated the same way as "failed". If it wouldn't be for my learning experience on bsc#990384 I would probably still see "failed" as more important than "softfailed with failed modules" but I think this is just a technical meaning, not any meaning for the product: Yes, we can revert from a snapshot and go on with testing other modules. But to me it does not bear a higher meaning regarding the quality of the SUT.

**Comment by coolo** [comment]
IMO we have 'half passes' and 'half fails' :)

File: templates/test/overview.html.ep (hunk 1 above)
Comment (coolo) [inline templates/test/overview.html.ep]: because i is HTML2 and has been deprecated since then

**Comment by coolo** [comment]
we just had a lengthy discussion about how to solve this intersex state - but this will be a bigger stunt, so I merged this one.

**Comment by Martchus** [comment]
Ok. And yes, the different softfails should be distinguished somehow on a higher level. Maybe just some flag in the Jobs scheme or an additional state.

**Pull request merged**